gp-nested-forms/gpnf-override-parent-merge-tag-on-submission.php

```diff
@@ -7,66 +7,156 @@
+				 */
+				$should_include = in_array( $child_field->id, $this->include_field_ids );
+				$should_exclude = in_array( $child_field->id, $this->exclude_field_ids );
+
+				if ( empty( $this->include_field_ids ) && $should_exclude ) {
+					continue;
+				}
+
+				if ( ! $should_include ) {
 					continue;
 				}
+
 				if ( $child_field->get_entry_inputs() ) {
-					foreach ( $child_field->get_entry_inputs() as $input ) {
-						gpnf_override_child_entry_input_value( $entry, $field, $input['id'], rgar( $input, 'defaultValue' ) );
-					}
+					$inputs = $child_field->get_entry_inputs();
 				} else {
-					gpnf_override_child_entry_input_value( $entry, $field, $child_field->id, $child_field->defaultValue );
+					$inputs = array(
+						array(
+							'id'           => $child_field->id,
+							'defaultValue' => $child_field->defaultValue,
+						),
+					);
+				}
+
+				foreach ( $inputs as $input ) {
+					$this->override_child_entry_input_value( $entry, $field, $input['id'], rgar( $input, 'defaultValue' ) );
 				}
 			}
 		}
+
+		return $entry;
 	}
 
-	return $entry;
-}
+	function override_child_entry_input_value( $entry, $field, $input_id, $default_value ) {
 
-function gpnf_override_child_entry_input_value( $entry, $field, $input_id, $default_value ) {
+		preg_match_all( '/{Parent:(\d+(\.\d+)?)[^}]*}/i', $default_value, $matches, PREG_SET_ORDER );
+		if ( empty( $matches ) ) {
+			return;
+		}
 
-	preg_match_all( '/{Parent:(\d+(\.\d+)?)[^}]*}/i', $default_value, $matches, PREG_SET_ORDER );
-	if ( empty( $matches ) ) {
-		return;
-	}
+		$value = $default_value;
+		foreach ( $matches as $match ) {
+			$value = str_replace( $match[0], rgar( $entry, $match[1] ), $value );
+		}
 
-	$value = $default_value;
-	foreach ( $matches as $match ) {
-		$value = str_replace( $match[0], rgar( $entry, $match[1] ), $value );
-	}
+		$child_entry_ids = explode( ',', rgar( $entry, $field->id ) );
+		foreach ( $child_entry_ids as $child_entry_id ) {
+			GFAPI::update_entry_field( $child_entry_id, $input_id, $value );
+		}
 
-	$child_entry_ids = explode( ',', rgar( $entry, $field->id ) );
-	foreach ( $child_entry_ids as $child_entry_id ) {
-		GFAPI::update_entry_field( $child_entry_id, $input_id, $value );
 	}
-
 }
 
-add_filter( 'gform_entry_post_save', 'gpnf_override_parent_merge_tags', 11, 2 );
+# -------------------------------------------------
+# Configuration Options:
+#
+# child_form_id      (int)   optional id of child form to apply this to. If excluded, then this will apply to all child forms
+# exclude_field_ids  (array) optional array of field IDs to exclude from the override.
+# included_field_ids (array) optional array of field IDs to apply this to. If included, then exclude_field_ids will be ignored.
+# -------------------------------------------------
+
+# Example - apply to all fields of a specific child form
+// new GPNF_Override_Parent_Merge_Tags(
+// 	array(
+// 		'child_form_id' => 1,
+// 	)
+// );
```

Comment:
Can you put the `array` on the same line as the class constructor call per feedback in that other PR? Someday we'll have a lint or Dangerfile rule for this 😅

Reply:
Done!

A dangerfile rule would be sick :fire: